share/mk/bsd.lib.mk

```diff
@@ -99,7 +99,7 @@ LDFLAGS+= -Wl,-zretpolineplt
 LDFLAGS.bfd+= -Wl,-znoexecstack
 .if ${MK_BRANCH_PROTECTION} != "no"
 CFLAGS+=  -mbranch-protection=standard
-.if ${MACHINE_ARCH} == "aarch64" && defined(BTI_REPORT_ERROR)
+.if ${LINKER_FEATURES:Mbti-report}
```

Comment:
I included the `defined(BTI_REPORT_ERROR)` check as I wasn't sure if it would break ports.

Reply:
I'll add that as well.

Reply:
@zxombie: Just, to confirm... `BTI_REPORT_ERROR` is not defined anywhere, do you want to have this check globally? (i.e. as part of the linker feature).

Reply:
It's passed in on the command line (or appropriate make config file) to enable the check. Until we are happy anything it could break has been fixed I was planning on keeping it off.

Reply:
Done. Thank you for clarifying.
- [x] Guards against old linkers (fixes cross-builds)
- [x] Turned into a linker feature
- [x] Activated by a `BTI_REPORT_ERROR` variable